src/impl_methods.rs

```diff
@@ -560,11 +560,11 @@ where
         (0..self.ndim()).for_each(|ax| {
```

Comment:
Sorry if too frivolous, but I missed before that style wise this would be preferred to be a regular for loop. Slightly more idiomatic and avoids creating a few more functions to codegen.

Reply:
Okay, no problem. I've added that change.

I guess I'm used to using `for_each` instead of `for` loops whenever possible because, for some iterators, `for_each` can be faster. For a range iterator like this, though, that's probably a premature optimization, and you're right that a `for` loop is more idiomatic and avoids the extra closure.

The docs say:

> It's generally more idiomatic to use a `for` loop, but `for_each` may be more legible when processing items at the end of longer iterator chains. In some cases `for_each` may also be faster than a loop, because it will use internal iteration on adaptors like `Chain`.

I wonder if the compiler could be changed to automatically implement `for` loops using `for_each` instead of `next` when possible.